sky/skylet/log_lib.py

```diff
@@ -377,7 +377,7 @@ def tail_logs(job_owner: str,
     log_path = os.path.join(log_dir, 'run.log')
     log_path = os.path.expanduser(log_path)
 
-    status = job_lib.query_job_status(job_owner, [job_id])[0]
+    status = job_lib.get_status(job_id)
```

Comment:
Why this change? IIUC, `query_job_status()` ensures latest statuses, while `get_status()` is looking up the DB, which may be stale?

Reply:
Ahh, that is a good point. I added the job status update back. During adding this, I realized that there could be a race condition between the job status query (or called update now), and the generated ray program. I added a filelock for each job, so that the job status update will be atomic. PTAL.